Add PySAL and Scikit-HEP domain stack metapackages (#381)
scientific-python/specs: core-projects/_index.md

## Description

Core Projects are depended upon by many of the other projects,
and often provide basic data structures, drawing primitives,

> implementations of fundamental algorithms, or are metapackages for a subecosystem
> representing a particular scientific field or domain.

Reviewer: ```suggestion
implementations of fundamental algorithms, or are metapackages
that represent a particular scientific field.
```

Probably don't need the double indirection.
